Marlin/src/module/probe.cpp

```diff
@@ -966,7 +959,7 @@ float Probe::probe_at_point(const_float_t rx, const_float_t ry, const ProbePtRai
     switch (raise_type) {
       default: break;
       case PROBE_PT_RAISE:
-        do_z_clearance(Z_CLEARANCE_BETWEEN_PROBES);
+        do_z_clearance_by((Z_PROBE_SAFE_CLEARANCE) + _MIN(probe.offset.z, 0));
```

Comment:
With a BLTouch probe, the `probe.offset.z` can be anything between - let's say -0.5mm and -5mm, since the deployed probe and the stowed probe can have that much difference. This is one of the advantages of BLTouch is that it has such a versatile working range, simplifying its mounting.

The `Z_PROBE_SAFE_CLEARANCE` is meant to give adequate clearance for lateral motion, and to be the least amount of clearance required for the current probe mode. BLTouch HS mode has the advantage that the probe re-deploys on its own before arriving at the next probe point. But since it requires more of a raise, it may actually lead to an increase in the overall probing time on machines with a slow Z axis.

The `raise_after` parameter tells `run_z_probe` what to do in preparation for the next probe, or if it is the last probe it tells `run_z_probe` to stow the probe. The change I made above to override `raise_after` and do a "hard stow" on the BLTouch when it's not in HS mode is meant to invoke the raise in `probe.stow`, which may be larger or smaller than `Z_CLEARANCE_BETWEEN_PROBES`. From a scan of the code it looks like BLTouch is happy to stow without any raise at all, so it may not raise at all outside of HS mode.

If this "trick" is problematic then we can get rid of the extra `raise_type` finagling above and just use `raise_after`.

In general, it might be cool to just make these methods receive exact Z values for their various raises and generalize probing, so a command like `G30` can be given overrides, and higher-level methods can pass in the user-configured values when doing pre-configured probing procedures. Something to contemplate for a future revision.

Reply:
In the mentioned case, if stow provides no lift, the lift will come from `do_blocking_move_to()`, when `probe_at_point()` is next called. It works, but doesn't seem ideal.

That said, stow does do a raise, but only if the following statement is false: `#if EITHER(FIX_MOUNTED_PROBE, NOZZLE_AS_PROBE) && DISABLED(PAUSE_BEFORE_DEPLOY_STOW)` (which is the case for BLTOUCH, so a lift is included)

https://github.com/MarlinFirmware/Marlin/blob/3cb855c7d21efbe85030add1234d8d3b1fb209a7/Marlin/src/module/probe.cpp#L508-L519

Zdest should be reworked a little, as the switch to absolute height means the `- zoffs` should be unnecessary.